### ISSUE TITLE
Documentation updates for PR #8

### ISSUE THREAD

**Issue by cstratopoulos** [boostorg/ptr_container#16]
This issue is in reference to https://github.com/boostorg/ptr_container/pull/8 which conditionally provides interfaces with `auto_ptr` or `unique_ptr` to work around `auto_ptr` deprecation with newer compilers.

If I understand correctly, this PR was too late for Boost 1.66 having been merged in late November, but it will be in Boost 1.67; moreover there is still time to submit documentation updates before the 1.67 release.

I noticed on the [1.67 beta snapshot documentation](https://www.boost.org/doc/libs/1_67_0_beta1/libs/ptr_container/doc/ptr_container.html) that `auto_ptr` is still uniformly referred to in documentation, with no mention of `unique_ptr`. 

I'm opening this issue just to bring attention to this matter, but also maybe it's appropriate to have a discussion on how to treat this situation. One thought I had was that in other projects it's common practice to use a special identifier like _`implementation-defined`_ in lieu of actual object types, with a documentation page explaining its meaning. Maybe `auto_ptr<T>` could be replaced with something like _`smart-pointer-type<T>`_, with a documentation page explaining that Boost.Config will be used to resolve this to either `auto_ptr<T>` or `unique_ptr<T>` depending on compiler standard? Maybe it would be appropriate to add this to an "Upgrading from Boost v. 1.66.*" heading, along the lines of similarly named headings on the documentation homepage.

**Comment by eldiener** [comment]
I appreciate your bringing this up. The doc is written in reStructuredText. I am not the original maintainer or else I would have written the doc using Boost quickbook. If you would care to create a PR with changes to the appropriate .rst files I would be glad to take a look at it and merge it in order to produce updated docs for the library.

**Comment by cstratopoulos** [comment]
@eldiener Understood, and thanks for getting back to me. Will try to get started on something on my lunch break or after work today, hopefully will have a PR ready sometime tomorrow.

**Comment by eldiener** [comment]
Take your time with the PR, there is no rush. You are correct about also committing the HTML files generated by the rst file changes.

**Comment by eldiener** [comment]
As long as you can maintain the possible creation of the html pages from the shell scripts I would not worry about how you do it. Your upgrade section is excellent and the name of 'compatible-smart-ptr' is fine and better than 'auto-or-unique-ptr'. But are you sure that the pointer container interface replaces auto_ptr with unique_ptr when both are available as opposed to adding unique_ptr as an alternative when both are available and replacing auto_ptr with unique_ptr only when auto_ptr is no longer available. I did not code those changes.

**Comment by cstratopoulos** [comment]
@eldiener That is a good catch--I just took a closer look at the way the changes are implemented, and the wording for `BOOST_NO_AUTO_PTR` and ` BOOST_NO_CXX11_SMART_PTR `. 

In the case of `auto_ptr` parameters it is exactly as you said:
> adding unique_ptr as an alternative when both are available and replacing auto_ptr with unique_ptr only when auto_ptr is no longer available.

And that can be seen [here](https://github.com/boostorg/ptr_container/pull/8/files#diff-73b042bfaf009e0eb712813022401185R106) for example.

When it comes to function return types such overloading is of course not possible, so we get a `unique_ptr` return precisely when `auto_ptr` is not available, as seen e.g.,  [here](https://github.com/boostorg/ptr_container/pull/8/files#diff-7688fcffa72357ad793e84c4a5399f6fR768)


Anyway I've finished making all the changes to individual files. Probably tomorrow I'll correct the upgrading section and update the compatible smart pointer page to reflect the parameter vs. return type subtlety, then go ahead an open the PR. Thanks again for your feedback thus far.
